### DISCUSSION TITLE
How to make expiration notifications?

### DISCUSSION THREAD

**Discussion by SaffronPolina** [netbox-community/netbox#7023] [Q&A] [answered]
### NetBox version

v2.11.4

### Feature type

New functionality

### Proposed functionality

Hello, I use NetBox for accounting of licenses too. 
So, licenceses have term of expiration, and I cannot find any way for making auto notifications (on email for example) about it.
Same problem with technical support.
Is any decisiong of this problem?

### Use case

Now I need to upload CSV file from NetBox every week - it`s uncomfortable.

**Comment by nicosalvadore** [comment, marked as answer]
@SaffronPolina 
As a work-around, you could use the netbox HTTP API to retrieve your licenses, check the expiration dates, and send you an email about it.
You could create a python script that would implement this logic quite easily I think.

**Reply by SaffronPolina** [reply]
Personally me not, my development team - of cause can, but I was hope that someone already make script or any other decision and we could use it) Thank you for your help, will keep looking for ready script, maybe I will find it.

**Reply by nicosalvadore** [reply]
@SaffronPolina I've written a quick python3 script to gather all devices that have in the "comments" field a string "licenses-expiration:YYYY-MM-DD".

https://gist.github.com/nicosalvadore/2b87fc74ea0b17ec6d642ca405048000

The email notification part is not implemented yet though.

**Reply by SaffronPolina** [reply]
Thank you! It`s very usefull!
